source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,582 @@
+
+### Mapping the rules
+
+Mapping is a two way game: it is needed to translate *from* a networking
+API security group rule *to* an OVN ACL, ***and*** from an OVN ACL to a
+networking API security group rule.
+
+To achieve the latter, the information that is not **directly** encoded
+in the ACL - e.g. protocol, ethertype, ports (both min & max), security
+group id, remote ip prefix, and description - will be duplicated in the
+ACL external IDs.
+
+As the software matures, (most of) this information can eventually be
+extracted from the ACL match clause, but, in the first implementation,
+the plan is to duplicate it.
+
+The security group rule **white list** behavior will be implemented
+through the priority concept; throughout the code two different
+priorities will be used: a lesser priority for **all** ACLs having a
+deny *action* and an higher priority for **all** ACLs having *allow* or
+*allow-related* actions.
+
+## Activating the feature
+
+The feature will be always active, and it is a matter of identifying the
+ports that will be subject to the security group filters.
+
+As previously mentioned, the **port-security** attribute will be used to
+know if a port will be added to the **deny-all** port group, having all
+IP traffic dropped as a consequence.
+
+The **port-security** attribute is set at port creation time, and its value will
+be inherited from the corresponding *network* entity when not defined at port
+level. Afterwards, it can be updated through the API - at port *or* network
+levels.
+
+The default behavior for the network's **port-security** attribute would be
+defined in the configuration file, through the ***port-security-enabled***,
+property, located in a new section, called **NETWORK**. This value will be set
+during *engine-setup* - refer to [this section](#engine-setup) - and the
+default choice will be 'yes', which will result in all traffic being dropped to
+*and* from the VMs.
+
+Existent ports can later be updated, disabling the **port-security** attribute, which would remove the ACLs that drop all IP traffic to and from the VM.
+
+As defined in the [network port security attribute definition](https://developer.openstack.org/api-ref/network/v2/#port-security), updating the **port-security** attribute on a network object **does not** cascade the value to ports attached to that network - meaning that effect would only apply to ports created *after* that update.
+
+## User workflow
+
+Since the feature is active by default, the intended workflow - from the oVirt
+engine user's perspective - is:
+
+* create an external network through the provider - its default *port_security_enabled* value is true.
+* create VMs attached to this external network - they will inherit the
+*port_security_attribute* from the network, which will result in all IP traffic
+to *and* from that VM being dropped.
```

Comment:
I don't think that dropping all traffic from VM is a good default

Reply:
I understand the concern, and actually even share it. I just went with the flow and mimicked networking-ovn behavior, which is to drop everything *if* port_security is enabled (which it is by default).

The question is a bit of mimicking the networking-ovn behavior versus surprising the user.

I would rather **not surprise** existing users - e.g. set the default to **false** - but would like @yanivlavi opinion on this matter.

Reply:
Sounds good